### ISSUE TITLE
Supporting bilateral net settlement

### ISSUE THREAD

**Issue by koekiebox** [mojaloop/project#3907]
## **Goal**:
`As a` <stakeholder|persona>

`I want to` <action>
  
`so that` <reason> 

**Acceptance Criteria**:
- [ ] TBD

**Complexity**: <High|Medium|Low> > A short comment to remind the reason for the rating

**Uncertainty**: <High|Medium|Low> > A short comment to remind the reason for the rating

---

**Tasks**:
- [ ] TBD [ @? ]

**Done**
- [ ] Acceptance Criteria pass
- [ ] Designs are up-to date
- [ ] Unit Tests pass
- [ ] Integration Tests pass
- [ ] Code Style & Coverage meets standards
- [ ] Changes made to config (default.json) are broadcast to team and follow-up tasks added to update helm charts and other deployment config.
- [ ] TBD

## **Pull Requests**:
- [ ] TBD

## **Follow-up**:
- N/A

**Dependencies**:
- N/A

## **Accountability**:
- Owner: TBC
- QA/Review: TBC

**Comment by PaulMakinMojaloop** [comment]
@koekiebox With regard to:
"Add settlementType to ISettlementConfig, with options: MULTILATERAL_DEFERRED_NET_SETTLEMENT and
BILATERAL_DEFERRED_NET_SETTLEMENT, as well as API support"
Can you expand on the API support you are referring to?

**Comment by PaulMakinMojaloop** [comment]
@koekiebox  If I'm understanding this correctly, an ISettlementBatchAccountBilateralMapping reflects the bilateral settlement position between two participants; and a settlement batch would/could consist of multiple instances of ISettlementBatchAccountBilateralMapping.

Then ISettlementBatchAccountBilateralMapping consists of two instances of IAccountBilateral, one for each participant in the bilateral position.

Each IAccountBilateral includes creditBalance, debitBalance and netBalance for that participant, with regard to the other participant.

Can I ask why we need all three balances? Wouldn't netBalance be sufficient?

Further, why do we need two instances of IAccountBilateral? Why not just have participant IDs, and a netBalance, with sign indicating which it is in favour of?

Sorry if I'm oversimplifying, i'm just trying to understand the rationale.

**Comment by koekiebox** [comment]
> @koekiebox With regard to: "Add settlementType to ISettlementConfig, with options: MULTILATERAL_DEFERRED_NET_SETTLEMENT and BILATERAL_DEFERRED_NET_SETTLEMENT, as well as API support" Can you expand on the API support you are referring to?

@PaulMakinMojaloop This would be the API updates necessary in order to store and retrieve the models as part of a settlement config. The endpoints include:
- GET `/models` - Fetch a settlement config by name
- GET `/models/:id` - Fetch a config by id
- POST `/models` - Create config

**Comment by koekiebox** [comment]
> Can I ask why we need all three balances? Wouldn't netBalance be sufficient?

@PaulMakinMojaloop We would be storing the debitBalance and creditBalance only. This is for tracking and traceability. It provides a clear picture of each balance. We would then calculate the netBalance on demand based on credit/debit.

**Comment by koekiebox** [comment]
> If I'm understanding this correctly, an ISettlementBatchAccountBilateralMapping reflects the bilateral settlement position between two participants; and a settlement batch would/could consist of multiple instances of ISettlementBatchAccountBilateralMapping.
> 
> Then ISettlementBatchAccountBilateralMapping consists of two instances of IAccountBilateral, one for each participant in the bilateral position.

Yes, this is correct. ✅

**Comment by koekiebox** [comment]
> Further, why do we need two instances of IAccountBilateral? Why not just have participant IDs, and a netBalance, with sign indicating which it is in favour of?
> 
> Sorry if I'm oversimplifying, i'm just trying to understand the rationale.

Not at all. This is something we can discuss. I would be in favour of the individual accounts, purely due to being able to see all of the debit/credit balances for each of the accounts, similar to what you brought up with regards to the netBalance. 
That being said, it would still be possible to lookup the transfers based on the "batchAccountId" in order to determine how the balance was calculated.
